File: common/src/liquidity.rs
use crate::{Field, Uint64};

/// The data structure that represents a user's liquidity in a pair of two specific tokens.
pub struct Liquidity {
    pub wdb_index: Field,
    pub base_token_id: Field,
    pub quote_token_id: Field,
    pub points: Uint64,
}

File: common-o1js/src/structs/liquidity.ts
import { Field, PublicKey, Struct, UInt64 } from "o1js"

/** The provable data structure that represents a user's liquidity inside a single AMM pool. */
export class Liquidity extends Struct({
    provider: PublicKey,
    baseTokenId: Field,
    quoteTokenId: Field,
    points: Field,
}) {
    toFields(): [Field, Field, Field, Field, Field] {
        return [
            this.provider.x,
            this.provider.isOdd.toField(),
            this.baseTokenId,
            this.quoteTokenId,
            this.points,
        ]
    }
}
